reports/report.md

```diff
@@ -1988,6 +1996,71 @@ It appears that the contract includes a payable function to accept Ether but lac
 
 
 
+## H-37: Incorrect ERC721 interface.
+
+Incorrect return values for ERC721 functions. A contract compiled with Solidity > 0.4.22 interacting with these functions will fail to execute them, as the return value is missing. Set the appropriate return values and types for the defined ERC721 functions.
+
+<details><summary>9 Found Instances</summary>
+
+
+- Found in src/IncorrectERC721.sol [Line: 14](../tests/contract-playground/src/IncorrectERC721.sol#L14)
+
+	```solidity
+	    function balanceOf(address owner) external view returns (uint72) {
+	```
+
+- Found in src/IncorrectERC721.sol [Line: 18](../tests/contract-playground/src/IncorrectERC721.sol#L18)
+
+	```solidity
+	    function ownerOf(uint256 tokenId) public view returns (bytes4) {
+	```
+
+- Found in src/IncorrectERC721.sol [Line: 22](../tests/contract-playground/src/IncorrectERC721.sol#L22)
+
+	```solidity
+	    function approve(
+	```
+
+- Found in src/IncorrectERC721.sol [Line: 37](../tests/contract-playground/src/IncorrectERC721.sol#L37)
+
+	```solidity
+	    function getApproved(uint256 tokenId) external view returns (uint72) {
+	```
+
+- Found in src/IncorrectERC721.sol [Line: 41](../tests/contract-playground/src/IncorrectERC721.sol#L41)
+
+	```solidity
+	    function setApprovalForAll(
+	```
+
+- Found in src/IncorrectERC721.sol [Line: 49](../tests/contract-playground/src/IncorrectERC721.sol#L49)
+
+	```solidity
+	    function isApprovedForAll(
+	```
+
+- Found in src/IncorrectERC721.sol [Line: 56](../tests/contract-playground/src/IncorrectERC721.sol#L56)
+
+	```solidity
+	    function transferFrom(
+	```
+
+- Found in src/IncorrectERC721.sol [Line: 71](../tests/contract-playground/src/IncorrectERC721.sol#L71)
+
+	```solidity
+	    function safeTransferFrom(
+	```
+
+- Found in src/IncorrectERC721.sol [Line: 103](../tests/contract-playground/src/IncorrectERC721.sol#L103)
+
+	```solidity
+	    function supportsInterface(
```

Comment:
This is not defined in ERC721. Should it be caught @TilakMaddy ?

Reply:
I had the same doubt but I saw that Slither codebase uses this ...so I decided to add it

Reply:
That's a bug IMO, this function is from ERC165. Let's only catch instances where 721 is incorrect!

Reply:
✅